### ISSUE TITLE
Symlink name replaced with original on push

### ISSUE THREAD

**Issue by einhander** [odeke-em/drive#183]
I have directory with symlinks with different names than original files.
On push drive want to chenge names of this files to original.
````
drive push  Студентам/
Resolving...
- /Студентам/Вопросы к экзамену по ЛХП.pdf
- /Студентам/Вопросы к экзамену по ТТПБД.pdf
- /Студентам/Образец отчета по лабораторным работам.pdf
- /Студентам/Методичка очистка и рекуперация.pdf
+ /Студентам/exam.pdf
+ /Студентам/lab_report_1.pdf
+ /Студентам/questions.pdf
+ /Студентам/recuperation.pdf
Deletion count 4 dest: 445.79KB
Addition count 4 src: 445.79KB
Proceed with the changes? [Y/n]:
````

Debian amd64, drive from github.

**Comment by odeke-em** [comment]
Hello @einhander thank you for reporting this and welcome to drive.
I can now see the pain that this will cause, I actually intentionally replaced the symlink resolved paths because in issues someone alluded to a case of duplicate paths being created because of some case I didn't understand where they had different names.
I introduced this in commit ffcc48ebe3994ae90a8fe00afa0ed85830ce3f2e
```diff
diff --git a/src/push.go b/src/push.go
index fce2fb8..8dbbc93 100644
--- a/src/push.go
+++ b/src/push.go
@@ -461,12 +461,16 @@ func list(context *config.Context, p string, hidden bool, ignore *regexp.Regexp)
...
                        symlink := (file.Mode() & os.ModeSymlink) != 0
-                       if symlink {
+
+                       if !symlink {
+                               resPath := gopath.Join(absPath, file.Name())
+                               fileChan <- NewLocalFile(resPath, file)
+                       } else {
                                symAbsPath := gopath.Join(absPath, file.Name())
                                var symResolvPath string
                                symResolvPath, err = filepath.EvalSymlinks(symAbsPath)
@@ -479,7 +483,7 @@ func list(context *config.Context, p string, hidden bool, ignore *regexp.Regexp)
                                if err != nil {
                                        continue
                                }
-                               fileChan <- NewLocalFile(symAbsPath, symInfo)
+                               fileChan <- NewLocalFile(symResolvPath, symInfo)
                        }
                }
                close(fileChan)
```

If you wouldn't mind, I'll push up a quick patch in a few seconds to revert this and if you could help me test it that'd be appreciated.

**Comment by odeke-em** [comment]
Hello again @einhander welcome to drive and thank you for reporting this issue!
Pardon my previous comments, for I narrowed the issue to the actual name swap from the resolved file.
Please take a look at PR #184

**Comment by odeke-em** [comment]
Addressed by PR #184, please get the latest and thank you for reporting this :)
